chunky/src/java/se/llbit/math/Ray.java

```diff
@@ -386,7 +387,17 @@ public void setPrevMaterial(Material mat, int data) {
 
   public void setCurrentMaterial(Material mat, int data) {
     this.currentMaterial = mat;
-    this.currentData = data;
+    if (data == 0) {
+      if (mat instanceof Water) {
+        this.currentData = ((Water) mat).data;
+      } else if (mat instanceof Lava) {
+        this.currentData = ((Lava) mat).data;
+      } else {
+        this.currentData = 0;
+      }
+    } else {
+      this.currentData = data;
+    }
```

Comment:
Does the ray really need to store the data anymore?

Reply:
The ray's data is used in the water octree traversal code and removing it causes bugs there. This keeps it compatible with the rest of Chunky until we refactor the way water works (i.e. not require the ray to know the current and previous water data).

Reply:
But can't the data be queried from the currentMaterial?

Reply:
Maybe, but that's for another PR

Reply:
I don't know, it's a bit weird in the water intersection and world intersect code. With some refactoring, `data` can be dropped in the future, but that's probably for another PR.